ember-server/shared/src/test/scala/org/http4s/ember/server/ConnectionSuite.scala

```diff
@@ -165,4 +179,34 @@ class ConnectionSuite extends Http4sSuite {
       } yield assertEquals(chunk, None)
   }
 
+  fixture().test("#6931 - bad request on request/start-line which can not be parsed") { client =>
+    val request = Stream(
+      "GET /hello/colt?foo={ HTTP/1.1\r\n\r\n"
+    )
+
+    for {
+      _ <- client.writes(fs2.text.utf8.encode(request))
+      resp <- client.responsePrelude
+    } yield assertEquals(resp.status, Status.BadRequest)
+  }
+
+  ResourceFunFixture(
+    emberServerBuilder(defaultIdleTimeout, defaultHeaderTimeout)
+      .withRequestLineParseErrorHandler { case _ =>
+        IO.pure(Response(Status.InternalServerError))
```

Comment:
Feels naughty to test something that shouldn't be allowed by spec 😉

Reply:
haha. I'll change it to 301

Reply:
@isomarcte Do you want to fix that test?

Reply:
It's not a big deal, let's not block on this any longer.

Reply:
Can we merge, then? Did @rossabaker or @ChristopherDavenport want to chime in at all on this?

Reply:
Yeah, I'm hoping for one more look at this :)

Reply:
Sorry, @armanbilge @coltfred, I've been short on time recently. I can update that test in another PR.